tests/integration/go/yaml/main.go

```diff
@@ -0,0 +1,49 @@
+package main
+
+import (
+	"path/filepath"
+
+	corev1 "github.com/pulumi/pulumi-kubernetes/sdk/v2/go/kubernetes/core/v1"
+	"github.com/pulumi/pulumi-kubernetes/sdk/v2/go/kubernetes/yaml"
+	"github.com/pulumi/pulumi/sdk/v2/go/pulumi"
+)
+
+func main() {
+	pulumi.Run(func(ctx *pulumi.Context) error {
+		_, err := yaml.NewConfigFile(ctx, "guestbook",
+			&yaml.ConfigFileArgs{File: "guestbook-all-in-one.yaml"},
+		)
+		if err != nil {
+			return err
+		}
+
+		resources, err := yaml.NewConfigGroup(ctx, "manifests",
+			&yaml.ConfigGroupArgs{
+				Files: []string{filepath.Join("manifests", "*.yaml")},
+				Transformations: []yaml.Transformation{
+					func(state map[string]interface{}, opts ...pulumi.ResourceOption) {
+						if state["apiVersion"] == "v1" && state["kind"] == "Pod" {
+							metadata := state["metadata"].(map[string]interface{})
+							_, ok := metadata["labels"]
+							if !ok {
+								metadata["labels"] = map[string]string{"foo": "bar"}
+							} else {
+								labels := metadata["labels"].(map[string]string)
+								labels["foo"] = "bar"
+							}
+						}
+					},
+				},
+			},
+		)
+		if err != nil {
+			return err
+		}
+
+		if resources != nil {
```

Comment:
How can this be `nil`?  I would not expect user code to need to do this check - is there a reason it does?

Reply:
You're right; the only way this could be `nil` is in case of an error, which should already be caught by checking the `err` variable.